File: app/src/main/java/com/xiaobai/game/mygame/httptest/NetWork.java
package com.xiaobai.game.mygame.httptest;

import android.content.Context;


/**
 * Created by lijingyin on 2017/12/4.
 */

public class NetWork {

    private static NetWork instatnce;

    private static YLApi ylApi;

    public NetWork(Context mContext) {
        ylApi = new YLApi(mContext);
    }

/*    public NetWork(Context mContext,boolean isTest) {
        ylApi = new YLApi(mContext,isTest);
    }*/
/*
    public static NetWork getTestInstance(Context mContext) {
        if (null == instatnce) {
            instatnce = new NetWork(mContext,true);
        }
        return instatnce;
    }
*/

    public static NetWork getInstance(Context mContext) {
        if (null == instatnce) {
            instatnce = new NetWork(mContext);
        }
        return instatnce;
    }

    public YLApi getYlApi() {
        return ylApi;
    }


}

File: app/src/main/java/com/xiaobai/game/mygame/http/NetWork.kt
package com.xiaobai.game.mygame.http

import android.content.Context

/**
 *  网络请求入口类
 * Created on 2018/5/15.
 * @author wr
 * @version 1.0.0
 */
 open class NetWork {

    private var api: API = API()

    /**
     * 获取 API实例
     */
    fun getAPI(): API {
        return api
    }

    companion object{
        private var instanced: Any?=null
        /**
         * 获取 NetWork实例
         */
        fun getInstance(mContext: Context) :NetWork{
            if (null == instanced) instanced = NetWork()
            return instanced as NetWork
        }
    }

}

